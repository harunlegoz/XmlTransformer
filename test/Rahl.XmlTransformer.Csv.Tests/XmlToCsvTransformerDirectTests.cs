using System.IO;
using System.Text;
using System.Threading.Tasks;
using ApprovalTests;
using NUnit.Framework;
using Rahl.XmlTransformer;
using Rahl.XmlTransformer.Tests.Core;

namespace Rahl.XmlTransformer.Csv.Tests
{
    public class XmlToCsvTransformerDirectTests
    {
        [Test]
        public async Task ShouldDirectTransformBasicXmlToCsvSuccessfully()
        {
            // Assign
            var config = new TransformerConfiguration();
            var transformer = new XmlToCsvTransformer(config);
            var xml = TestData.Basic1;

            // Act
            var output = await transformer.TransformAsync(xml);

            // Assert
            Approvals.Verify(output);
        }

        [Test]
        public async Task ShouldDirectTransformComplexXmlToCsvSuccessfully()
        {
            // Assign
            var config = new TransformerConfiguration();
            var transformer = new XmlToCsvTransformer(config);
            var xml = TestData.Complex1;

            // Act
            var output = await transformer.TransformAsync(xml);

            // Assert
            Approvals.Verify(output);
        }

        [Test]
        public async Task ShouldDirectTransformArrayXmlToCsvSuccessfully()
        {
            // Assign
            var config = new TransformerConfiguration();
            var transformer = new XmlToCsvTransformer(config);
            var xml = TestData.Basic2;

            // Act
            var output = await transformer.TransformAsync(xml);

            // Assert
            Approvals.Verify(output);
        }

        [Test]
        public async Task ShouldDirectTransformArrayWithAttributesXmlToCsvSuccessfully()
        {
            // Assign
            var config = new TransformerConfiguration();
            var transformer = new XmlToCsvTransformer(config);
            var xml = TestData.Basic3;

            // Act
            var output = await transformer.TransformAsync(xml);

            // Assert
            Approvals.Verify(output);
        }

        [Test]
        public async Task ShouldDirectTransformComplexArrayXmlToCsvSuccessfully()
        {
            // Assign
            var config = new TransformerConfiguration();
            var transformer = new XmlToCsvTransformer(config);
            var xml = TestData.Complex2;

            // Act
            var output = await transformer.TransformAsync(xml);

            // Assert
            Approvals.Verify(output);
        }

        [Test]
        public async Task ShouldDirectTransformComplexArrayWithAttributesXmlToCsvSuccessfully()
        {
            // Assign
            var config = new TransformerConfiguration();
            var transformer = new XmlToCsvTransformer(config);
            var xml = TestData.Complex3;

            // Act
            var output = await transformer.TransformAsync(xml);

            // Assert
            Approvals.Verify(output);
        }
    }
}