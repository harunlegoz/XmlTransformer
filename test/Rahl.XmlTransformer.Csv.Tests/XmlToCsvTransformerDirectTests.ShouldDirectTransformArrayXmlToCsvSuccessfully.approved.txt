"_value"
"A"
"B"
"C"
"D"
"E"
